src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -0,0 +1,1050 @@
+                //     public int P1 { get; readonly set; } // ERR_AutoSetterCantBeReadOnly
+                Diagnostic(ErrorCode.ERR_AutoSetterCantBeReadOnly, "set").WithArguments("S.P1.set").WithLocation(4, 35),
+                // (6,51): error CS1604: Cannot assign to 'field' because it is read-only
+                //     public int P3 { get => field; readonly set => field = value; } // No ERR_AutoSetterCantBeReadOnly, but ERR_AssgReadonlyLocal
+                Diagnostic(ErrorCode.ERR_AssgReadonlyLocal, "field").WithArguments("field").WithLocation(6, 51),
+                // (7,42): error CS1604: Cannot assign to 'field' because it is read-only
+                //     public int P4 { get; readonly set => field = value; } // No ERR_AutoSetterCantBeReadOnly, but ERR_AssgReadonlyLocal
+                Diagnostic(ErrorCode.ERR_AssgReadonlyLocal, "field").WithArguments("field").WithLocation(7, 42)
+            );
+        }
+
+        [Fact]
+        public void TestAccessingNonExistingFieldMember()
+        {
+            var comp = CreateCompilation(@"
+public class C
+{
+}
+
+public class C2
+{
+    public int P1 { get => C.field; }
+    public int P2 { get => this.field; }
+}
+");
+            comp.VerifyDiagnostics(
+                // (8,30): error CS0117: 'C' does not contain a definition for 'field'
+                //     public int P1 { get => C.field; }
+                Diagnostic(ErrorCode.ERR_NoSuchMember, "field").WithArguments("C", "field").WithLocation(8, 30),
+                // (9,33): error CS1061: 'C2' does not contain a definition for 'field' and no accessible extension method 'field' accepting a first argument of type 'C2' could be found (are you missing a using directive or an assembly reference?)
+                //     public int P2 { get => this.field; }
+                Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "field").WithArguments("C2", "field").WithLocation(9, 33)
+            );
+        }
+
+        [Fact]
+        public void TestIndexer_LikeAutoProperty_ErrorCase()
+        {
+            var comp = CreateCompilation(@"
+public class C
+{
+    public int this[int i]
+    {
+        get; set;
+    }
+}
+");
+            comp.VerifyDiagnostics(
+                // (6,9): error CS0501: 'C.this[int].get' must declare a body because it is not marked abstract, extern, or partial
+                //         get; set;
+                Diagnostic(ErrorCode.ERR_ConcreteMissingBody, "get").WithArguments("C.this[int].get").WithLocation(6, 9),
+                // (6,14): error CS0501: 'C.this[int].set' must declare a body because it is not marked abstract, extern, or partial
+                //         get; set;
+                Diagnostic(ErrorCode.ERR_ConcreteMissingBody, "set").WithArguments("C.this[int].set").WithLocation(6, 14)
+            );
+        }
+
+        [Fact]
+        public void TestIndexer_ContainsFieldIdentifier_Error()
```

Comment:
>TestIndexer_ContainsFieldIdentifier_Error [](http://example.com/codeflow?start=20&length=41)

All tests should verify that we do not synthesize fields unexpectedly. Success cases can do this by observing emitted metadata. Note that private symbols are filtered out by default, either type IL should be checked, or compilation options should be adjusted to disable the filtering. Error cases should check ```GetFieldToEmit``` after checking diagnostics, but before checking ```NumberOfPerformedAccessorBinding```.